### PR TITLE
Updated multilevel modeling to v4

### PR DESCRIPTION
**Pull request by fonnesbeck**
Fixed models to work in v4.0.0b2 and added missing remote images as local files. 

I don't like the new formulation of the early models as having independent means for floor and basement, but I can tackle that issue in a separate PR.

This should supersede #282

### CODE REVIEW

**Comment by fonnesbeck** [comment]
<p>Sorry, not sure what you mean by decoy coordinates. Do they not have to be in the <code>coords</code> dict to be used by <code>floor_idx</code>? </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Unfortunately it is not compatible.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Unfortunately this does not seem to work. A <code>posterior_predictive</code> group does not get added does not get added to the trace and has to be added manually.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by OriolAbril** [comment]
<p>no, <code>floor_idx</code> is already an array of the right shape, so dims here have only 2 functions (out of 3 possible functions): annotating the dimension, and checking the shape matches for all objects annotated with that dimension. It is not being used to <em>define</em> the shape like it would in <code>pm.Normal("var", 0, 1, dims="obs_id")</code>. </p><p> </p><p>With the improved Aesara capabilities and the refactors to shape/size/dims functionality done by Brandon and Michael (I think it was mostly the two of them) when annotating dimensions, we can use dimension names that are not in <code>coords</code> which is very convenient when used in conjunction with <code>pm.Data</code> because the dimension and the variable where it is first annotated are linked so when generating predictions we only need to update the data instead of both data and coordinate values. See for example <a href="https://github.com/pymc-devs/pymc/blob/main/pymc/tests/test_shape_handling.py#L336" target="_blank">https://github.com/pymc-devs/pymc/blob/main/pymc/tests/test_shape_handling.py#L336</a>      </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by OriolAbril** [comment]
<p>this should add a <code>predictions</code> group, it seems to be working on the example in the docs: <a href="https://docs.pymc.io/en/latest/learn/examples/posterior_predictive.html#prediction" target="_blank">https://docs.pymc.io/en/latest/learn/examples/posterior_predictive.html#prediction</a>. Which groups do you have?</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by cfonnesbeck** [comment]
<p>Increased tuning seems to fix it.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by cfonnesbeck** [comment]
<pre class="ql-syntax" spellcheck="false">ValueError: Can not extract posterior_predictive from Inference data with groups:
	&gt; posterior
	&gt; predictions
	&gt; log_likelihood
	&gt; sample_stats
	&gt; observed_data
	&gt; constant_data
	&gt; predictions_constant_data! See /home/fonnesbeck/miniforge3/envs/pymc-dev-py39/lib/python3.9/site-packages/arviz/data/converters.py for other conversion utilities.
</pre>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by cfonnesbeck** [comment]
<p>OK, figured it out.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
OK, should be good to go now, unless there are further issues.

**Comment by OriolAbril** [comment]
/pre-commit run

Trying to see if this fixes the ci job, you'll have to pull before doing the changes.

**Comment by OriolAbril** [comment]
I never can get pre-commit magic comment to work :shrug:

**Comment by fonnesbeck** [comment]
<p>Is there anything we can do about this now?</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by OriolAbril** [comment]
<p>use the <code>model.set_data</code> method two times or speed https://github.com/pymc-devs/pymc/pull/5588 up </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by cfonnesbeck** [comment]
<p>Let's punt on this until <code>set_data</code> is updated. </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/285/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
Should be clean now.

File: myst_nbs/case_studies/multilevel_modeling.myst.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ kernelspec:
 
 # A Primer on Bayesian Methods for Multilevel Modeling
```
Comment (OriolAbril) [inline myst_nbs/case_studies/multilevel_modeling.myst.md:14]: The myst target before the title is missing: https://docs.pymc.io/en/latest/contributing/jupyter_style.html#first-cell

File: myst_nbs/case_studies/multilevel_modeling.myst.md [hunk 2]
```diff
@@ -1149,7 +1134,3 @@ mcelreath2018statistical
 %load_ext watermark
 %watermark -n -u -v -iv -w
 ```
```
Comment (OriolAbril) [inline myst_nbs/case_studies/multilevel_modeling.myst.md:1136]: The page footer is also missing https://docs.pymc.io/en/latest/contributing/jupyter_style.html#epilogue

**Review by OriolAbril — COMMENTED** [review]
commented on the myst because it was easier, best to make the changes on from jupyter

**Comment by fonnesbeck** [comment]
I'm getting an erroneous jupytext failure:

image

The two links in this paragraph are external.

**Comment by OriolAbril** [comment]
These should be flagged, it's the check name that is not right, it finds "urls that should be sphinx cross-references"

The one to the arviz homepage should be `` {doc}`arviz:index` `` the one to the xarrayforarviz notebook should be `` {ref}`xarray_for_arviz` ``

File: examples/case_studies/multilevel_modeling.ipynb [hunk 3]
```diff
@@ -4,10 +4,11 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "(notebook_name)=\n",
     "# A Primer on Bayesian Methods for Multilevel Modeling\n",
     "\n",
     ":::{post} 27 February, 2022\n",
-    ":tags: hierarchical, pymc.Data, pymc.Deterministic, pymc.Exponential, pymc.LKJCholeskyCov, pymc.Model, pymc.MvNormal, pymc.Normal\n",
+    ":tags: hierarchical, Data, Deterministic, Exponential, LKJCholeskyCov, Model, MvNormal, Normal\n",
```
Comment (OriolAbril) [inline examples/case_studies/multilevel_modeling.ipynb:11]:
```suggestion
    ":tags: hierarchical model, case study\n",
```

File: examples/case_studies/multilevel_modeling.ipynb [hunk 4]
```diff
@@ -4,10 +4,11 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "(notebook_name)=\n",
```
Comment (OriolAbril) [inline examples/case_studies/multilevel_modeling.ipynb:7]:
```suggestion
    "(multilevel_modeling)=\n",
```

File: myst_nbs/case_studies/multilevel_modeling.myst.md [hunk 5]
```diff
@@ -6,15 +6,16 @@ jupytext:
     format_version: 0.13
     jupytext_version: 1.13.7
 kernelspec:
-  display_name: Python 3 (ipykernel)
+  display_name: Python 3.9.10 ('pymc-dev-py39')
   language: python
   name: python3
 ---
 
+(notebook_name)=
```
Comment (OriolAbril) [inline myst_nbs/case_studies/multilevel_modeling.myst.md:14]:
```suggestion
(multilevel_modeling)=
```

File: myst_nbs/case_studies/multilevel_modeling.myst.md [hunk 6]
```diff
@@ -6,15 +6,16 @@ jupytext:
...
 # A Primer on Bayesian Methods for Multilevel Modeling
 
 :::{post} 27 February, 2022
-:tags: hierarchical, pymc.Data, pymc.Deterministic, pymc.Exponential, pymc.LKJCholeskyCov, pymc.Model, pymc.MvNormal, pymc.Normal
+:tags: hierarchical, Data, Deterministic, Exponential, LKJCholeskyCov, Model, MvNormal, Normal
```
Comment (OriolAbril) [inline myst_nbs/case_studies/multilevel_modeling.myst.md:18]:
```suggestion
:tags: hierarchical model, case study
```

**Review by OriolAbril — APPROVED** [review]

**Comment by OriolAbril** [comment]
related to #47

**Comment by hectormz** [comment]
@fonnesbeck I know this is an old PR, but I was wondering if you made the model pngs added in this PR via latex or some other method? Thanks!

**Pull request merged**